starter-core/src/main/java/io/micronaut/starter/openrewrite/OpenRewriteFeature.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2017-2022 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.starter.openrewrite;
+
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.starter.feature.Feature;
+
+/**
+ * A feature backed by an OpenRewrite recipe.
+ * @author Sergio del Amo
+ */
+public interface OpenRewriteFeature extends Feature {
+
+    @NonNull
+    String getRecipeName();
```

Comment:
I suppose this also supports using [Refaster and imperative recipes](https://docs.openrewrite.org/authoring-recipes/types-of-recipes)?

Reply:
Yes, I have tried only with declarative recipes. But ideally, we should support every type of recipe.

Reply:
maybe we could use a default convention such as io.micronaut.starter.feature.${featureName} to ensure unique name